modules/apps/message-boards/message-boards-web/src/main/resources/META-INF/resources/message_boards/js/MBPortlet.es.js

```diff
@@ -12,225 +12,243 @@
  * details.
  */
 
-import {PortletBase, fetch} from 'frontend-js-web';
-import core from 'metal';
-import {EventHandler} from 'metal-events';
+import {fetch} from 'frontend-js-web';
 
 const RECENTLY_REMOVED_ATTACHMENTS = {
 	multiple: Liferay.Language.get('x-recently-removed-attachments'),
 	single: Liferay.Language.get('x-recently-removed-attachment'),
 };
 
+const CONFIRM_DISCARD_IMAGES = Liferay.Language.get(
+	'uploads-are-in-progress-confirmation'
+);
+
 /**
  * MBPortlet handles the actions of replying or editing a
  * message board.
- *
- * @abstract
- * @extends {PortletBase}
  */
 
-class MBPortlet extends PortletBase {
-
-	/**
-	 * @inheritDoc
-	 */
-
-	created() {
-		this.eventHandler_ = new EventHandler();
+export default function MBPortlet({
```

Comment:
One particularity of using `liferay-frontend:component` is that it will do a `new MBPortlet.default(...)`.

As such, you could export a `Class` here so you could use class members, even private ones that seems that could be useful below...

Reply:
It would also allow to add the a `dispose` method to make sure all those event listeners are properly removed.